src/diffusers/schedulers/scheduling_dpmsolver_multistep.py

```diff
@@ -164,6 +164,7 @@ def __init__(
         lower_order_final: bool = True,
         euler_at_final: bool = False,
         use_karras_sigmas: Optional[bool] = False,
+        final_sigmas_type: Optional[str] = "default",  # "denoise_to_zero", "default"
```

Comment:
Are both options necessary? Each has an advantage in certain scenarios?

Reply:
* `default` is what we currently have:
I want to point out that there is some inconsistency between `use_karras_sigmas = True` and `use_karras_sigmas = False`: with karras_sigmas, we are skipping the last denoising step. I'm not entirely sure why it is done this way so I'm hoping to address that in a new PR so we don't delay this one getting merged in 

 current code 
```python
            sigmas = np.flip(sigmas).copy()
            sigmas = self._convert_to_karras(in_sigmas=sigmas, num_inference_steps=num_inference_steps)
            timesteps = np.array([self._sigma_to_t(sigma, log_sigmas) for sigma in sigmas]).round()
            sigmas = np.concatenate([sigmas, sigmas[-1:]]).astype(np.float32)
```

I think it should be 
```python
            sigmas = np.flip(sigmas).copy()
            sigmas = self._convert_to_karras(in_sigmas=sigmas, num_inference_steps=(num_inference_steps+1))
            timesteps = np.array([self._sigma_to_t(sigma, log_sigmas) for sigma in sigmas]).round()
```

https://github.com/huggingface/diffusers/blob/49a4b377c1fa5d1d5c6ebb3e480967db0b753297/src/diffusers/schedulers/scheduling_dpmsolver_multistep.py#L266

* `denoise_to_zero` matches the last step in k-diffusion. We can apply this with or without the karras sigmas. see some of my comparison results here https://github.com/huggingface/diffusers/pull/6477#issuecomment-1892830959

Reply:
I'm wondering if we should change the default to `"denoise_to_zero"` here though if you think it always gives better results. Also cc @sayakpaul

Reply:
we can do that! I will deprecate the `dpmsolver` and `dpmsolver-sde` since they won't work with the `denoise_to_zero` and I think they are not really used at all

Reply:
Ok for me!